components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -124,9 +94,36 @@ export const useSwap = () => {
   const [useDirectRoute, setUseDirectRoute] = useState<boolean>(false)
   const [slippageTolerance, setSlippageTolerance] = useState<string>('0.5')
   const [selectedGasFeeOption, setSelectedGasFeeOption] = useState<GasFeeOption>(gasFeeOptions[1])
-  const [initialized, setInitialized] = useState<boolean>(false)
-  const [tokenBalances, dispatchBalancesUpdate] = useReducer(balancesReducer, {})
-  const [abortController, setAbortController] = useState<AbortController | undefined>(undefined)
+
+  const tokensForBalances = useMemo(() => {
+    if (!selectedNetwork || !selectedAccount || !assetsList) {
+      return []
+    }
+
+    if (selectedNetwork.coin !== selectedAccount.accountId.coin) {
+      return []
+    }
+
+    if (selectedNetwork.coin === BraveWallet.CoinType.SOL) {
```

Comment:
might want to add a comment here explaining why we dont need to pass tokens for SOL

Reply:
Fixed here: [4c8463e](https://github.com/brave/brave-core/pull/19337/commits/4c8463e324659917fd666c7f1ae8bd85e25130b6)